modules/core/src/main/kotlin/io/github/lmliam/kotventure/core/selector/parsing/SelectorSnbtParsing.kt

```
@@ -0,0 +1,116 @@
+        else -> {
+            val key = readWhile(Char::isAllowedInUnquotedSelectorToken)
+            if (key.isEmpty()) fail("Expected SNBT compound key")
+        }
+    }
+}
+
+private fun SelectorReader.readSnbtUnquotedScalar(): String {
+    val start = offset
+    while (true) {
+        val character = peek() ?: break
+        if (character == ',' || character == '}' || character == ']' || character.isWhitespace()) break
```

detekt (1.23.8)

[warning] This condition is too complex (4). Defined complexity threshold for conditions is set to '4'

(detekt.complexity.ComplexCondition)